Blaise.Tests.Helpers/Browser/BrowserHelper.cs

```diff
@@ -79,5 +90,100 @@ private static ChromeDriver CreateChromeDriver()
 
             return new ChromeDriver(BrowserConfigurationHelper.ChromeDriver, chromeOptions);
         }
+
+        //AP changes - 20-03-2023
+        public static IWebElement WaitForElement(By by, bool checkIfClickable = false, int maxAttempts = 50)
+        {
+            IWebElement element = null;
+            var attempts = 0;
+
+            while (element == null && attempts < maxAttempts)
+            {
+                try
+                {
+                    element = _browser.FindElement(by);
+                }
+                catch (Exception)
+                {
+                    /*Ignored - continue polling*/
+                }
+
+                if (element != null && IsElementVisible(element))
+                {
+                    if (checkIfClickable)
+                    {
+                        if (IsElementClickable(element))
+                        {
+                            break;
+                        }
+                    }
+                    else
+                    {
+                        break;
+                    }
+                }
+                else
+                {
+                    System.Threading.Thread.Sleep(500);
+                }
+
+                attempts++;
+            }
+
+            if (element == null)
```

Comment:
This `if` statement currently isn't doing anything. If it's not used, can we removed it?

Reply:
Have added an exception on the back if the state of the element